File: data/stageConfig.ts
import { StageCfg } from '../types';

/**
 * Stage configuration with 20 pet forms
 * Progression: Ocean -> Land -> Final form
 * Thresholds increase difficulty as specified
 */
export const STAGE_CONFIG: StageCfg = {
  stages: [
    { i: 0, name: "Tadpole", threshold: 10, asset: "pet_tadpole.png" },
    { i: 1, name: "Minnow", threshold: 20, asset: "pet_minnow.png" },
    { i: 2, name: "Frog", threshold: 30, asset: "pet_frog.png" },
    { i: 3, name: "Hermit Crab", threshold: 40, asset: "pet_hermit.png" },
    { i: 4, name: "Starfish", threshold: 50, asset: "pet_starfish.png" },
    { i: 5, name: "Jellyfish", threshold: 60, asset: "pet_jellyfish.png" },
    { i: 6, name: "Squid", threshold: 75, asset: "pet_squid.png" },
    { i: 7, name: "Seahorse", threshold: 90, asset: "pet_seahorse.png" },
    { i: 8, name: "Dolphin", threshold: 110, asset: "pet_dolphin.png" },
    { i: 9, name: "Shark", threshold: 135, asset: "pet_shark.png" },
    { i: 10, name: "Otter", threshold: 165, asset: "pet_otter.png" },
    { i: 11, name: "Fox", threshold: 200, asset: "pet_fox.png" },
    { i: 12, name: "Lynx", threshold: 240, asset: "pet_lynx.png" },
    { i: 13, name: "Wolf", threshold: 285, asset: "pet_wolf.png" },
    { i: 14, name: "Bear", threshold: 335, asset: "pet_bear.png" },
    { i: 15, name: "Bison", threshold: 390, asset: "pet_bison.png" },
    { i: 16, name: "Elephant", threshold: 450, asset: "pet_elephant.png" },
    { i: 17, name: "Rhino", threshold: 515, asset: "pet_rhino.png" },
    { i: 18, name: "Lion", threshold: 585, asset: "pet_lion.png" },
    { i: 19, name: "Floating God", threshold: 0, asset: "pet_god.png" } // Final stage, no further evolution
  ]
};

/**
 * Get stage by index
 */
export function getStage(index: number) {
  return STAGE_CONFIG.stages[index];
}

/**
 * Get all stage names for debugging/display
 */
export function getAllStageNames(): string[] {
  return STAGE_CONFIG.stages.map(stage => stage.name);
}

/**
 * Check if stage is final (no further evolution)
 */
export function isFinalStage(index: number): boolean {
  return index === STAGE_CONFIG.stages.length - 1;
}

File: ios-native/SharedKit/Sources/SharedKit/StageConfig.swift
import Foundation

public struct StageConfig: Codable {
    public struct Entry: Codable {
        public let name: String
        public let threshold: Int

        public init(name: String, threshold: Int) {
            self.name = name
            self.threshold = threshold
        }
    }

    public let stages: [Entry]

    public init(stages: [Entry]) {
        self.stages = stages
    }

    /// Default configuration with 16 stages and increasing thresholds
    public static func defaultConfig() -> StageConfig {
        return StageConfig(stages: [
            Entry(name: "pet_baby", threshold: 0),      // Stage 0
            Entry(name: "pet_toddler", threshold: 10),   // Stage 1
            Entry(name: "pet_frog", threshold: 25),      // Stage 2
            Entry(name: "pet_hermit", threshold: 45),    // Stage 3
            Entry(name: "pet_seahorse", threshold: 70),  // Stage 4
            Entry(name: "pet_dolphin", threshold: 100),  // Stage 5
            Entry(name: "pet_alligator", threshold: 135), // Stage 6
            Entry(name: "pet_beaver", threshold: 175),   // Stage 7
            Entry(name: "pet_wolf", threshold: 220),     // Stage 8
            Entry(name: "pet_bear", threshold: 270),     // Stage 9
            Entry(name: "pet_bison", threshold: 325),    // Stage 10
            Entry(name: "pet_elephant", threshold: 385), // Stage 11
            Entry(name: "pet_rhino", threshold: 450),    // Stage 12
            Entry(name: "pet_adult", threshold: 520),    // Stage 13
            Entry(name: "pet_ceo", threshold: 595),      // Stage 14
            Entry(name: "pet_gold", threshold: 675)      // Stage 15
        ])
    }
}
